src/rdiff_backup/connection.py

```diff
@@ -588,7 +594,11 @@ def _extract_exception(self):
                 ),
                 log.INFO,
             )
-        return sys.exc_info()[1]
+        result = sys.exc_info()[1]
+        # OSError code are specificto the platform. Send back code as string.
+        if isinstance(result, OSError) and result.errno:
+            result.errno_str = errno.errorcode.get(result.errno, "EUNKWN")
```

Comment:
```suggestion
            result.errno_orig = result.errno
```
Just to facilitate a potential cross-platform debugging session, so that it's clear if a conversion of errno happened.